packages/Meshfree/test/PointCloudProblemGenerator/PointCloudProblemGenerator.hpp

```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_POINTCLOUDPROBLEMGENERATOR_HPP
+#define DTK_POINTCLOUDPROBLEMGENERATOR_HPP
+
+#include "DTK_ConfigDefs.hpp"
+#include "DTK_Types.h"
+
+#include <Kokkos_View.hpp>
+
+namespace DataTransferKit
+{
+//---------------------------------------------------------------------------//
+template <class Scalar, class SourceDevice, class TargetDevice>
+class PointCloudProblemGenerator
+{
+  public:
+    virtual ~PointCloudProblemGenerator() = default;
+
+    /*!
+     * \brief Create a problem where all points are uniquely owned (i.e. no
+     * ghosting)
+     *
+     * \param field_dim The number of components in the field.
+     *
+     * \param src_coords Coordinates of the source points. Layout:
+     * (point,dim).
+     *
+     * \param src_field Field values at the source points. A field may have
+     * multiple components. A generator implementation may choose to fill this
+     * with meaningful field data or may choose to initialize it to
+     * zero. Layout: (point,comp).
```

Comment:
Documentation got out of sync.  Also, why did you remove the field values all together?  I recall in previous review only asking for `field_dim` parameter to be removed.

Reply:
So the big issue with the field being returned by the generator is also providing the expected result on the target points. Do we return the expected target result? 

I would argue the correct result depends on the operator one uses. So, for example, the nearest operator correct result is different from a standard interpolation. 

On the opposite side, one could argue that these generator tests just check "how close" we are to some expected solution. The nearest neighbor test would then check the results relative to some large tolerance.

@dalg24 what do you think?

Reply:
Generator should at the very least allocate memory and returns properly sized view for the target values.  When it is an option it should fill the view with expected data.  In the context of testing, then it will let us assert that the answer is close enough.